qlib/log.py

```diff
@@ -165,8 +165,16 @@ def filter(self, record):
         return allow
 
 
+@contextmanager
```

Comment:
@zhupr   Please keep the previous  `set_global_logger_level` function and create a new `contextmanager`.
I think they are both very useful.
Besides, please add detailed docstrings about the feature. 

Thanks